src/handlers/create-hold.ts

```diff
@@ -0,0 +1,46 @@
+import { RouteHandler } from "fastify";
+import { createItem } from "../services/item-generators";
+import { isRequestable } from "../services/item-predicates";
+import HoldsStore from "../services/HoldsStore";
+import { PatronHoldPost } from "../types/patrons";
+
+type UrlParams = {
+  patronId: string;
+};
+
+export const createHold =
+  (
+    holdsStore: HoldsStore
+  ): RouteHandler<{ Params?: UrlParams; Body?: PatronHoldPost }> =>
+  async (request, reply) => {
+    const { patronId } = request.params || {};
+    if (!patronId || !request.body?.recordNumber) {
+      reply.code(400).send({
+        code: 115,
+        specificCode: 0,
+        httpStatus: 400,
+        name: "Invalid JSON",
+        description: "JSON object missing field or field has invalid data",
+      });
+      return reply;
+    }
+
+    const itemId = request.body!.recordNumber.toString();
```

Comment:
I've just read about the non-null assertion operator – didn't know it existed. Neat.

Reply:
Makes! code! more! exciting!